_posts/2019-01-29-OAS-3.md

```diff
@@ -0,0 +1,91 @@
+OpenAPI 3 is the new version of the OpenAPI Specification, the successor to
+OpenAPI 2 (or formerly known as Swagger 2).
+
+< FIXME Add screenshot of OAS 3 doc in Apiary using Apiary UI >
+
+While we are not yet supporting all of the new features added OAS 3 such as
+[callbacks](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#callbackObject)
+and
+[links](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#linkObject)
+, we've built a solid foundation of our parser which will allow us to
+incrementally roll out support for new features across our entire product line.
+Whether it be in our mock server, [Dredd](https://dredd.org/en/latest/), our
+[new documentation renderer](http://blog.apiary.io/ApiaryUI) or our [API Style
+Guide](https://help.apiary.io/tools/style-guide/) product.
+
+Our OpenAPI 3 parser is designed with user experience in mind. It brings
+significant improvements over our OpenAPI 2 parser.
+
+### Better Warning and Error Reporting
+
+The most significant difference is the parser is able to emit much clearer
+warnings and errors when the OpenAPI syntax is not semantically correct. This
+was a large source of user frustration for our OpenAPI 2 users. With OpenAPI 2
+you'd often get hard to understand warnings or errors due to the OpenAPI 2
+parser relying on the JSON Schema from OpenAPI Initiative as the base of
+validations. JSON Schema validations often produce unclear error messages.
+
+We've taken validation completely into our own hands so we can have far greater
+control over the user experience.
+
+As an example, if you miss a required `in` property for a Parameter Object in
+OpenAPI 2, you'll often see a confusing error such as the following:
+
+> Data does not match any schemas from 'oneOf'
+
+You'll see the exact same error message if the value of `in` is not a permitted
+value or is not a string.
+
+Whereas for similar semantic issues, our OpenAPI 3 parser will emit clearer
+errors:
+
+> 'Parameter Object' is missing required property 'in'
+
+> 'Parameter Object' 'in' must be either 'query, 'header', 'path' or 'cookie'"
+
+The state of the errors and warnings in our OpenAPI 2 parser can also fall down
+further when you are dealing with referencing as the entire OAS 2 document is
+dereferenced (stripped of `$ref`) up front, which means that the context of an
+error can be missing and it can be hard to pin-point the exact line which
+caused a parsing error.
+
+### Validation Speed
+
+Another key user experience improvement will be with the performance of our new
+parser in comparison to our OAS 2 parser. We can parse extremely large OAS 3
+documents (multiple megabytes) in an order of magnitude faster than we could
+for OAS 2 so you can get much quicker feedback in our editor.
+
+### Try OpenAPI 3.0 Today
+
+You can try OpenAPI 3.0 today in both Apiary and our open source tools such as
+[Dredd](https://github.com/apiaryio/api-elements.js/issues/71).
```

Comment:
This will look like a link leading to dredd.org, so as a reader I'd ignore it. I'd be more verbose with something like

```
such as Dredd - see example usage in our [GitHub issue #71](...)
```

Also, there is no mention of how one can try it out in Apiary now/today (further down there's how to try it out later in the development process). Do they need to have the new ApiaryUI turned on? Is it just sufficient to paste the OAS3 document into the editor?

Reply:
> It says "in this stage of rollout" (now):

I guess never mind. I thought about it and I think it boils down how we Czechs do not distinguish the difference between _this_ and _that_ sufficiently.